modules/core/src/test/java/org/apache/ignite/internal/processors/security/cache/closure/ContinuousQueryWithTransformerRemoteSecurityContextCheckTest.java

```diff
@@ -0,0 +1,136 @@
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test RemoteFilterFactory of ContinuousQueryWithTransformer. */
+    @Test
+    public void testRemoteFilterFactory() {
+        Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> consumer =
+            new Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>>() {
+                @Override public void accept(ContinuousQueryWithTransformer<Integer, Integer, Integer> q) {
+                    q.setRemoteFilterFactory(new TestRemoteFilterFactory());
+                    q.setRemoteTransformerFactory(new TestTransformerFactory(false));
+                }
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test TransformerFactory of ContinuousQueryWithTransformer. */
+    @Test
+    public void testTransformerFactory() {
+        Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> consumer =
+            new Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>>() {
+                @Override public void accept(ContinuousQueryWithTransformer<Integer, Integer, Integer> q) {
+                    q.setRemoteTransformerFactory(new TestTransformerFactory(true));
+                }
+            };
+
+        runAndCheck(grid(SRV_INITIATOR), operation(consumer));
+        runAndCheck(grid(CLNT_INITIATOR), operation(consumer));
+    }
+
+    /** Test operation. */
+    private IgniteRunnable operation(Consumer<ContinuousQueryWithTransformer<Integer, Integer, Integer>> c) {
+        return () -> {
+            VERIFIER.register();
+
+            ContinuousQueryWithTransformer<Integer, Integer, Integer> cq = new ContinuousQueryWithTransformer<>();
+
+            cq.setLocalListener(e -> {/* No-op. */});
+
+            c.accept(cq);
+
+            openQueryCursor(cq);
+        };
+    }
+
+    /** Test transformer factory. */
+    private static class TestTransformerFactory implements Factory {
+        /** True if closure should be registered in verifier. */
+        private final boolean register;
+
+        /** Constructor. */
+        public TestTransformerFactory(boolean register) {
+            this.register = register;
+        }
+
+        /** {@inheritDoc} */
+        @Override public IgniteClosure<CacheEntryEvent, Object> create() {
+            return new TestIgniteClosure(register);
+        }
+    }
+
+    /** Test IgniteClosure. */
+    private static class TestIgniteClosure implements IgniteClosure<CacheEntryEvent, Object> {
+        /** True if closure should be registered in verifier. */
+        private final boolean register;
+
+        /** Calling of closure should be registered one time only. */
+        private final AtomicBoolean executed = new AtomicBoolean(false);
+
+        /** Constructor. */
```

Comment:
Convert to multi-line javadoc according to coding guidelines, add @param register Register flag.